modules/android/vsandroid_vcxproj.lua

```diff
@@ -314,9 +380,15 @@
 	premake.override(vc2010, "exceptionHandling", function(oldfn, cfg, condition)
 		if cfg.system == p.ANDROID then
 			-- Note: Android defaults to 'off'
-			if cfg.exceptionhandling then
+			if cfg.exceptionhandling and cfg.exceptionhandling ~= "Default" then
```

Comment:
Instead of adding this check, you could instead check `if exceptions[cfg.exceptionhandling] ~= nil then` before calling `vc2010.element`. You might even be able to remove this whole if statement, but you would need to add a check for the non-VS2015 version: `if cfg.exceptionhandling == premake.ON then`.

Reply:
```lua
local exceptions = {
				On = "Enabled",
				Off = "Disabled",
				UnwindTables = "UnwindTables",
			}
			if exceptions[cfg.exceptionhandling] ~= nil then
				if _ACTION >= "vs2015" then
					vc2010.element("ExceptionHandling", condition, exceptions[cfg.exceptionhandling])
				else
					if cfg.exceptionhandling == premake.ON then
						vc2010.element("GccExceptionHandling", condition, "true")
					end
				end
			end```

I hope this is good enough :)

Reply:
https://github.com/ifarbod/premake-core/commit/6b06e408e69fb56770860fb26335ab2ac68697a2

Reply:
It's a tad wonky having the action check inside the LUT check, but it's fine.

Reply:
I hope this is better: https://github.com/ifarbod/premake-core/commit/d8742697d345325fccf0ec8eb445c0698439b65b

Reply:
is the Off entry even necessary here as it matches the default?

also @samsinsane what about `p.ON` instead of `"On"` ?

with `iif` it then boils down to that:

```lua
if _ACTION >= "vs2015" then
	local exception = iif(cfg.exceptionhandling == p.ON, "Enabled", cfg.exceptionhandling)
	if exception then
		vc2010.element("ExceptionHandling", condition, exception)
	end
else
	if cfg.exceptionhandling == p.ON then
		vc2010.element("GccExceptionHandling", condition, "true")
	end
end
```

Reply:
@WorldofBay I like the `iif` approach more :)

Reply:
> is the Off entry even necessary here as it matches the default?

Yes, if a user explicitly specifies `Off` it should turn it off. Today the default is off, but tomorrow the default could easily be on. The user knows what they need, we shouldn't try to get all fancy here.

> also @samsinsane what about `p.ON` instead of `"On"`?

Not sure what you're referencing?

> with `iif` it then boils down to that:

The snippet doesn't work with two of the four possible values for `exceptionhandling`, and same as the first point, the user knows what they want.